rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -928,12 +929,17 @@ def process(self, message: Message, **kwargs: Any) -> None:
 
             message.set(ENTITIES, entities, add_to_output=True)
 
+        if out and DIAGNOSTIC_DATA in out:
+            message.add_diagnostic_data(
+                self.unique_name,
```

Comment:
We need to store the component index in the component itself, so we can give it a unique readable name for the diagnostic data output.